### PR TITLE
Refactored to track all changes from the queryController as they come…

### PR DESCRIPTION
**Pull request by daniel-maness**
… in instead of modifying the tableView. Then, when the queryController finishes with its changes, update the tableView with all changes at once.

### CODE REVIEW

File: Code/Controllers/ATLConversationListViewController.m [hunk 1]
```diff
@@ -536,6 +541,30 @@ - (void)queryControllerDidChangeContent:(LYRQueryController *)queryController
     }
 }
 
+- (NSMutableArray *)insertedRowIndexPaths
+{
+    if (_insertedRowIndexPaths == nil) {
+        _insertedRowIndexPaths = [[NSMutableArray alloc] init];
+    }
+    return _insertedRowIndexPaths;
```
Comment (jpmcglone) [inline Code/Controllers/ATLConversationListViewController.m]: how do you feel about 
```
return _insertedRowIndexPath ?: (_insertedRowIndexPaths = [[NSMutableArray alloc] init]);
```

Comment (daniel-maness) [inline Code/Controllers/ATLConversationListViewController.m]: Yeah much cleaner, I like it. Will update.

**Comment by jwyld** [comment]
This is essentially what we did with Cola, but we abstracted to a "change set" object which held all the changes.  We were able to reuse this for the conversation as well as the conversation list.  In Atlas, the code has gone through a number of iterations along this similar code path.  I'm not sure if we're seeing failures there.  If not, I would ask why not.  If we are seeing crashes, we should apply similar treatment there as well.

**Comment by LukeDurrant** [comment]
Nice!

**Pull request merged**